src/components/pages/species/show/fields/field/index.tsx

```diff
@@ -0,0 +1,130 @@
+import { Box, Stack } from "@chakra-ui/layout";
+import { SpeciesField } from "@interfaces/custom";
+import { SPECIES_FIELD_DELETED, SPECIES_FIELD_UPDATED } from "@static/events";
+import React, { useEffect, useState } from "react";
+import { useListener } from "react-gbus";
+import { useImmer } from "use-immer";
+
+import useSpecies from "../../use-species";
+import SpeciesTraitView from "../traits";
+import useSpeciesFields from "../use-species-field";
+import DocumentsField from "./documents";
+import SpeciesFieldHeading from "./heading";
+import OccuranceRecoardSpeciesField from "./occurance-records";
+import ReferencesField from "./references";
+import SpeciesFieldSimple from "./simple";
+import SpeciesFieldSimpleCreate from "./simple/create";
+
+interface SpeciesFieldGroupProps extends SpeciesField {
+  valueCallback;
+  level: number;
+}
+
+export default function SpeciesFieldGroup({
+  parentField,
+  childField,
+  valueCallback,
+  level
+}: SpeciesFieldGroupProps) {
+  const { showHiddenFields } = useSpeciesFields();
+  const [childFieldsHasValue, setChildFieldsHasValue] = useState<boolean>();
+  const { permissions } = useSpecies();
+
+  const [currentField, setCurrentField] = useImmer<any>(parentField);
+
+  // subscription for created or updated field
+  useListener(
+    (field) => {
+      if (currentField.id === field.fieldId) {
+        setCurrentField((_draft) => {
+          const idx = _draft.values.findIndex((o) => o.id === field.id);
+          if (idx !== -1) {
+            _draft.values[idx] = field;
+          } else {
+            _draft.values.unshift(field);
+          }
+        });
+      }
+    },
+    [SPECIES_FIELD_UPDATED]
+  );
+
+  // subscription for deleted field
+  useListener(
+    (field) => {
+      if (currentField.id === field.fieldId) {
+        setCurrentField((_draft) => {
+          const idx = _draft.values.findIndex((o) => o.id === field.id);
+          if (idx !== -1) {
+            _draft.values.splice(idx, 1);
+          }
+        });
+      }
+    },
+    [SPECIES_FIELD_DELETED]
+  );
+
+  useEffect(() => {
+    if (
+      (currentField.values.length || currentField.traits.length || childFieldsHasValue) &&
+      valueCallback
+    ) {
+      setChildFieldsHasValue(true);
+      valueCallback(true);
+    }
+  }, [currentField, childFieldsHasValue]);
+
+  // References
+  if (parentField.id === 81) {
```

Comment:
can below line be put in switch case format

Reply:
@teamdraftbox done!